src/sys/prctl.rs

```diff
@@ -213,3 +215,10 @@ pub fn set_thp_disable(flag: bool) -> Result<()> {
 pub fn get_thp_disable() -> Result<bool> {
     prctl_get_bool(libc::PR_GET_THP_DISABLE)
 }
+
+/// Set an identifier (or reset it) to the address memory range.
+pub fn set_addr_name(addr: NonNull<c_void>, length: NonZeroUsize, name: &CStr) -> Result<()> {
```

Comment:
I kinda think we should rename this function to `set_vma_anon_name()` so that it is consistent with the attribute name  `SET_VMA_ANON_NAME`, though I do agree that `set_addr_name` is the clearer one.